src/libs/auth.js

```diff
@@ -82,19 +82,30 @@ const revokeTokens = async () => {
 }
 
 const getSigninArgs = includeBillingScope => {
-  return includeBillingScope === true ? {
-    scope: `
-    openid
-    email
-    profile
-    https://www.googleapis.com/auth/cloud-billing
-  `.trim().split(/\s+/).join(' ')
-  } : {}
+  return Utils.cond(
+    [includeBillingScope === false, () => {}],
+    // For Google just append the scope to the signin args.
+    [isGoogleAuthority(), () => ({ scope: 'openid email profile https://www.googleapis.com/auth/cloud-billing' })],
+    // For B2C switch to a dedicated policy endpoint configured for the GCP cloud-billing scope.
+    () => ({
+      extraQueryParams: { access_type: 'offline', p: getConfig().b2cBillingPolicy },
+      extraTokenParams: { p: getConfig().b2cBillingPolicy }
+    })
+  )
 }
 
-export const signIn = (includeBillingScope = false) => {
+export const signIn = async (includeBillingScope = false) => {
   const args = getSigninArgs(includeBillingScope)
-  return getAuthInstance().signinPopup(args).catch(() => false)
+  const user = await getAuthInstance().signinPopup(args)
+
+  // For B2C record in the auth store whether we requested the GCP cloud-billing scope since there
+  // is no way to determine it after the fact.
+  // For Google we don't need to do this since we can inspect the scope directly in the user object.
+  if (!isGoogleAuthority()) {
```

Comment:
I actually thought about this and tried it. I decided against it because Google does some magic -- sometimes even if I don't initially sign in with the billing scope I have it _anyway_ (maybe from a prior session). So I thought it was better to test the actual value from Google.

For B2C there is no magic because they are distinct signin endpoints. So the state value is pretty reliable. 

(Btw it's probably possible to make B2C pass the Google scopes as a claim so the client could inspect them directly instead of using our own state -- but this works well enough I think.)